Completed Task "Command pages: Management" (#23)
MoonlightCapital/MoonlightBot-docs: management-commands/reaction-roles.md

# reaction-roles

This command allows you to set up reaction roles in your server. Reaction roles allow users to assign/remove roles
to/from themselves by reacting to specific messages that server admins set up. There are several customization options

> to optimize your needs.

Reviewer: explain this is a technical documentation for the command, and link to the tutorial

to/from themselves by reacting to specific messages that server admins set up. There are several customization options
to optimize your needs.

{% hint style="warning" %}
This is only the technical documentation of the command and not intended to be a tutorial on how to set up reaction
roles. For a tutorial on how to set up reaction roles, refer to the

> [Setting up reaction roles](../start-up/setting-up-reaction-roles.md) guide.

Reviewer: "reaction roles" repeated too many times imo

* Manage Server

For more information on editing permission requirements for specific users/roles, refer to
the [permissions tutorial](<linkToPermissionsTutorial>)

## set

> This command allows you to apply a role to an emoji in a group, if the group is not set up, it will be created.

Reviewer: ```suggestion
This command allows you to pair a role with an emoji in a group, if the group is not set up, it will be created.
```

specify: to unpair use the same command again; multiple roles can be paired to the same emoji and will be all given

## info

> This command allows you to view information about a group of reaction roles.

Reviewer: specify what info is shown

## autorepair

> This command automatically repairs errors in a group.

Reviewer: "repair" means removing the IDs of roles that do not exist anymore from the pairings

```text
/reaction-roles autorepair <group> [disconnect-messages]
```

* `group`: The name of the group to be repaired

> * `disconnect-messages`: Removes the group form all messages applied to (True/False)

Reviewer: this is a disruptive operation and makes reaction roles on messages this group was applied to non-functional. they need to be reapplied appropriately afterwards.

right click message => remove reaction roles is always the preferred option, unless the messages were deleted and becomes impossible, that is

> * `disconnect-messages`: Removes the group form all messages applied to (True/False)

Reviewer: ```suggestion
* `disconnect-messages`: Removes the group from all messages applied to (True/False)
```

## autorepair

This command automatically "repairs" errors in a group. With "repair" is meant that it will remove the IDs of roles that
do not exist anymore from the pairings.

> {% hint style="warning" %}

Reviewer: ```suggestion
{% hint style="danger" %}
```

as per contributing guide, this changes color to red
